### PR TITLE
[ET-1079] Make use of `is_singular` to support pages

### PR DESCRIPTION
**Pull request by rafsuntaskin**
### 🎫 Ticket

[ET-1079] 

### 🗒️ Description
* is_single doesn't work for `pages` type so switched to `is_singular`

### 🎥 Artifacts 
N/A

### ✔️ Checklist
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1079]: https://theeventscalendar.atlassian.net/browse/ET-1079

### CODE REVIEW

**Review by juanfra — APPROVED** [review]

File: src/Tribe/Admin/Manager/Service_Provider.php [hunk 1]
```diff
@@ -138,7 +138,7 @@ public function add_attendees_view_button() {
 		// If this is not the Admin Edit page or not the singular frontend view, bail out.
 		if (
 			! ( is_admin() && 'edit' == tribe_get_request_var( 'action' ) )
-			&& ! is_single()
+			&& ! is_singular()
```
Comment (juanfra) [inline src/Tribe/Admin/Manager/Service_Provider.php:141]: Should we get the array of CPTs supporting tickets and use as parameter?

File: src/Tribe/Admin/Manager/Service_Provider.php [hunk 2]
```diff
@@ -138,7 +138,7 @@ public function add_attendees_view_button() {
 		// If this is not the Admin Edit page or not the singular frontend view, bail out.
 		if (
 			! ( is_admin() && 'edit' == tribe_get_request_var( 'action' ) )
```
Comment (juanfra) [inline src/Tribe/Admin/Manager/Service_Provider.php:140]: Let's use strict comparison.

**Review by sc0ttkclark — APPROVED** [review]

File: src/Tribe/Admin/Manager/Service_Provider.php (hunk 2 above)
Comment (sc0ttkclark) [inline src/Tribe/Admin/Manager/Service_Provider.php:140]:
```suggestion
			! ( is_admin() && 'edit' === tribe_get_request_var( 'action' ) )
```

File: src/Tribe/Admin/Manager/Service_Provider.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Tribe/Admin/Manager/Service_Provider.php:141]: That could be a good idea to prevent the tickets queries running.

**Review by sc0ttkclark — CHANGES_REQUESTED** [review]
Some tweaks

File: src/Tribe/Admin/Manager/Service_Provider.php [hunk 3]
```diff
@@ -151,6 +151,12 @@ public function add_attendees_view_button() {
 			return;
 		}
 
+		$supported_post_types = tribe_get_option( 'ticket-enabled-post-types', [] );
```
Comment (bordoni) [inline src/Tribe/Admin/Manager/Service_Provider.php]:
```suggestion
		$supported_post_types = (array) tribe_get_option( 'ticket-enabled-post-types', [] );
```

**Review by bordoni — CHANGES_REQUESTED** [review]
Lets make sure we have a Changelog.

**Review by bordoni — APPROVED** [review]

File: src/Tribe/Admin/Manager/Service_Provider.php [hunk 4]
```diff
@@ -151,6 +151,12 @@ public function add_attendees_view_button() {
 			return;
 		}
 
+		$supported_post_types = (array) tribe_get_option( 'ticket-enabled-post-types', [] );
```
Comment (juanfra) [inline src/Tribe/Admin/Manager/Service_Provider.php:154]: I think this could go directly as a parameter of `is_singular`

Comment (sc0ttkclark) [reply src/Tribe/Admin/Manager/Service_Provider.php:154]: Yes, exactly what I thought too, is_singular takes a string or an array of strings for post types to limit by

**Review by sc0ttkclark — APPROVED** [review]

File: src/Tribe/Admin/Manager/Service_Provider.php (hunk 4 above)
Comment (sc0ttkclark) [reply src/Tribe/Admin/Manager/Service_Provider.php]: Added

File: src/Tribe/Admin/Manager/Service_Provider.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/Tribe/Admin/Manager/Service_Provider.php]: Added

**Comment by sc0ttkclark** [comment]
Note: No changelog needed as this code was added in this sprint

**Pull request merged**